src/saving/SaveHelper.cs

```diff
@@ -265,6 +265,37 @@ public static (int Count, ulong DiskSpace) CountSaves(string? nameStartsWith = n
         return (count, totalSize);
     }
 
+    /// <summary>
+    ///   Counts the total number of saves that have an older version than the
+    ///   current version of Thrive and are a backup and how many bytes they take up
+    /// </summary>
+    public static (int Count, ulong DiskSpace) CountOldBackupSaves()
+    {
+        int count = 0;
+        ulong totalSize = 0;
+
+        using var file = new File();
+        foreach (var save in CreateListOfSaves())
+        {
+            var saveInfo = global::Save.LoadJustInfoFromSave(save);
+
+            if (SaveUpgrader.IsSaveABackup(save) &&
+                VersionUtils.Compare(saveInfo.ThriveVersion, Constants.Version) < 0)
+            {
+                if (file.Open(Path.Combine(Constants.SAVE_FOLDER, save), File.ModeFlags.Read) != Error.Ok)
+                {
+                    GD.PrintErr("Can't read size of save file: ", save);
+                    continue;
+                }
+
+                ++count;
+                totalSize += file.GetLen();
+            }
+        }
+
```

Comment:
I personally would merge this with the other CountSaves() method. 
Maybe you tried to keep the logic of the other method clean but i usually try to keep the code lines as few as possible.

Reply:
Also maybe update the xml comment of that function to say what kind of saves it can count if u merge.

Reply:
I just didn't feel like messing all that much with the function because it would likely mean adding some sort of boolean parameter to clean old save file.

Reply:
it could be like 
`if (nameStartsWith == null || save.StartsWith(nameContains, StringComparison.CurrentCulture || (nameContains.Contains( ".backup" ) &&  SaveUpgrader.IsSaveABackup(save) &&  VersionUtils.Compare(saveInfo.ThriveVersion, Constants.Version) < 0 ) )`

Reply:
Purely from memory I seem to remember we already had a save counting method, so it would make sense to adjust it slightly to be able to count old backups rather than needing a new method that looks almost the same.

I'd rework the conditions to be opposite, to fit in the extra logic to check if the name ends with something instead. I don't both options should be usable at the same time. And separate out the counting outside the if, and just leave it there to set a boolean flag to make it clean. Very rough code:

```
            if (nameStartsWith != null && !save.StartsWith(nameStartsWith, StringComparison.CurrentCulture))
            {
                 continue
             }
             if(requiredSuffix != null && doesn't match suffix)
                continue;

             if (file.Open(Path.Combine(Constants.SAVE_FOLDER, save), File.ModeFlags.Read) != Error.Ok)
              {
                    GD.PrintErr("Can't read size of save file: ", save);
                    continue;
                }

                ++count;
               totalSize += file.GetLen();

```

Reply:
It might to adjust the parameters, instead of `string? nameStartsWith = null` I don't see why this couldn't be `string? nameContains = null` which does what it says on the tin

Reply:
That is not the same. If someone writes in their save in a random spot the text that identifies the different save types, then the save counting would not work. I was lazy when initially writing this so I just checked the name prefixes and assumed the save type number would be after that.
I guess this might be a case where regex is suitable (changing the method to take in a nullable regex object as a parameter), that way both cases could be properly handled.

Reply:
you can also name the save "quick_save_1", wouldnt that also break the count?

Reply:
Yeah, but I didn't consider it a problem if the player intentionally named saves like that as it would be pretty unlikely they do so by accident. Including the word "backup" anywhere in the save name is much more likely.

Reply:
I think its ".backup",meh, to mee it seems unlikely

Reply:
Would it be possible to allocate a separate directory for backups

Reply:
This should be fine to see if it contains `Constants.SAVE_BACKUP_SUFFIX` because it is defined as `public const string SAVE_BACKUP_SUFFIX = ".backup" + SAVE_EXTENSION_WITH_DOT;` which is `.backup.` and the player shouldn't be able to save files containing a dot

Reply:
> Yeah, but I didn't consider it a problem if the player intentionally named saves like that as it would be pretty unlikely they do so by accident. Including the word "backup" anywhere in the save name is much more likely.

would it be worth implementing the same suffix functionality for quick saves and auto saves, I get it's not a critical feature but it would remove this and I would be happy to work on it.

Reply:
I see regex as the only thing that can support both cases in a flexible enough way.

Reply:
> which is .backup. and the player shouldn't be able to save files containing a dot


As far as I know, there's nothing preventing the player from putting as many `.`s in the filename as they want.

Reply:
Shouldn't we just make the dot an illegal character for the name as a save file, or at least remove from the name for creating the savefile

Reply:
Alright, so I just need to adjust DeleteSaveOfType to take a regex expression instread

Reply:
I'll go with "yes", not having recently read through all of the save counting and deleting code so I can't exactly remember.

Reply:
I understand regex expressions need to be compiled, would it be better to take precompiled regex expression of type `Regex` or take a or take a valid regex expression with type `String`? It may be better to store these regex expressions as constants and as the `Regex` type as to avoid recompiling the expression

Reply:
100% take in a Regex object to just need to compile the regex once and not each time the method is called, or even worse, for each different save that is checked.